frontend/components/common/InstantSearchProvider.tsx

```diff
@@ -149,12 +149,9 @@ export function InstantSearchProvider({
          qsModule: typeof qs;
          location: Location;
       }) {
-         const pathParts = location.pathname
-            .split('/')
-            .filter((part) => part !== '');
-         const firstPathSegment = pathParts.length >= 1 ? pathParts[0] : '';
-         const deviceHandle = pathParts.length >= 2 ? pathParts[1] : '';
-         const itemType = pathParts.length >= 3 ? pathParts[2] : '';
+         const { firstPathSegment, deviceHandle, itemType } = getRouteData(
+            location.pathname
+         );
```

Comment:
This is definitely for later:

This "common" file has custom parsing for certain types of pages.
We should stop doing it his way and instead all each page to pass in a function.

Simplified Current code:

```
<InstantSearchProvider {...props}>
	<DefaultLayout>
		{children} /* Used on every page */
	</DefaultLayout>
</InstantSearchProvider>
```

My claim:

```
	<DefaultLayout>
		{children} /* Used on every page */
	</DefaultLayout>
```

MyPage.tsx
```jsx
const createUrl = () => {...};
const parseUrl = () => {...};
const props = useInstantSearchProps({createUrl, parseUrl});

<InstantSearchProvider {...props}>
	<Text>
		Hello
	</Text>
</InstantSearchProvider>
```

Reply:
hmm seems reasonable but i just want to get this pull out rn